tests/virtual_machines_test.go

```diff
@@ -0,0 +1,389 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kubevirtv1 "kubevirt.io/kubevirt/pkg/api/v1"
+)
+
+var _ = Describe("Virtual Machines", func() {
+
+	BeforeAll(func() {
+		result := testClient.KubeClient.ExtensionsV1beta1().RESTClient().
+			Post().
+			RequestURI(fmt.Sprintf(nadPostUrl, TestNamespace, "linux-bridge")).
+			Body([]byte(fmt.Sprintf(linuxBridgeConfCRD, "linux-bridge", TestNamespace))).
+			Do()
+		Expect(result.Error()).NotTo(HaveOccurred())
+	})
+
+	Context("Check the client", func() {
+		AfterEach(func() {
+			vmList := &kubevirtv1.VirtualMachineList{}
+			err := testClient.VirtClient.List(context.TODO(), &client.ListOptions{}, vmList)
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, vmObject := range vmList.Items {
+				err = testClient.VirtClient.Delete(context.TODO(), &vmObject)
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			Eventually(func() int {
+				vmList := &kubevirtv1.VirtualMachineList{}
+				err := testClient.VirtClient.List(context.TODO(), &client.ListOptions{}, vmList)
+				Expect(err).ToNot(HaveOccurred())
+				return len(vmList.Items)
+
+			}, 30*time.Second, 3*time.Second).Should(Equal(0), "failed to remove all vm objects")
+		})
+
+		Context("When the client wants to create a vm", func() {
+			It("should create a vm object and automatically assign a static MAC address", func() {
+				err := setRange(rangeStart, rangeEnd)
+				Expect(err).ToNot(HaveOccurred())
+
+				vm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br", "")},
+					[]kubevirtv1.Network{newNetwork("br")})
+
+				Eventually(func() error {
+					return testClient.VirtClient.Create(context.TODO(), vm)
+
+				}, 40*time.Second, 5*time.Second).Should(Not(HaveOccurred()), "failed to apply the new vm object")
+				_, err = net.ParseMAC(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+		//2166
```

Comment:
Move this comment to the test level please
//2166
Context("When the MAC address is within range", func() {